src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodec.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2017 Rocana
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.rocana.lucene.codec.v1;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.PostingsFormat;
+import org.apache.lucene.codecs.lucene54.Lucene54Codec;
+import org.apache.lucene.codecs.perfield.PerFieldPostingsFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Custom codec that speeds up opening Searchers.
+ *
+ * We don't need Lucene's entire-file checksum since HDFS already
+ * uses checksums for HDFS blocks. Eliminating Lucene's checksum for
+ * the entire file improves Searcher open performance.
+ *
+ * See ROCANA-8229
+ */
+public class RocanaSearchCodec extends FilterCodec {
+
+  public static final String NAME = "RocanaSearchCodecV1";
```

Comment:
Lucene does both: `org.apache.lucene.codecs.lucene54.Lucene54Codec`:
unique package and unique class name. 

Since Java packages are mostly "invisible" in code from a readability standpoint (meaning you have to go to the import statements to see them), I vote that we follow the Lucene model, since anyone using this library will likely be very familiar with Lucene and changing from the standard model needs a DGR, which I'm not sure we have.